File: 374. Guess Number Higher or Lower/javascript/main.js
/**
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 */
var guess = function (num) {
  if (num > 6) {
    return -1;
  } else if (num < 6) {
    return 1;
  } else return 0;
};

/**
 * @param {number} n
 * @return {number}
 */
var guessNumber = function (n) {
  let low = 1;
  let high = n;
  while (low <= high) {
    // Prevents potential integer overflow
    let middle = low + Math.floor((high - low) / 2);
    let result = guess(middle);
    if (result === 0) {
      return middle;
    } else if (result === 1) {
      low = middle + 1;
    } else {
      high = middle - 1;
    }
  }
};

console.log(guessNumber(10));

File: 374. Guess Number Higher or Lower/python/main.py
def guess(num: int) -> int:
    # This is a placeholder for the actual guess function
    # The actual implementation will be provided in the problem statement
    return 1

def guessNumber(n: int) -> int:
    low = 1
    high = n
    while low <= high:
        middle = low + (high - low) // 2
        result = guess(middle)
        if result == 0:
            return middle
        elif result == 1:
            low = middle + 1
        else:
            high = middle - 1
            
